src/overlays/actors/ovl_En_Rd/z_en_rd.c

```diff
@@ -21,71 +41,869 @@ const ActorInit En_Rd_InitVars = {
+
+    if (this->actor.params == 3) {
+        this->actor.flags |= 0x80;
+    }
+}
+
+void EnRd_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnRd* this = THIS;
+
+    if (gSaveContext.unk_1422 != 0) {
+        gSaveContext.unk_1422 = 0;
+    }
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void func_80AE2630(GlobalContext* globalCtx, Actor* thisx, s32 arg2) {
+    Actor* enemyIt = globalCtx->actorCtx.actorList[ACTORTYPE_ENEMY].first;
+
+    while (enemyIt != NULL) {
+        if ((enemyIt->id != ACTOR_EN_RD) || (enemyIt == thisx) || (enemyIt->params < 0)) {
+            enemyIt = enemyIt->next;
+            continue;
+        }
+
+        if (arg2 != 0) {
+            enemyIt->parent = thisx;
+        } else if (enemyIt->parent == thisx) {
+            enemyIt->parent = NULL;
+        }
+        enemyIt = enemyIt->next;
+    }
+}
+
+void func_80AE269C(EnRd* this) {
+    if (this->actor.params != 2) {
+        SkelAnime_ChangeAnimTransitionRepeat(&this->skelAnime, &D_060087D0, -6.0f);
+    } else {
+        SkelAnime_ChangeAnimDefaultRepeat(&this->skelAnime, &D_06005D98);
+    }
+
+    this->unk_31B = 0;
+    this->unk_30C = (Math_Rand_ZeroOne() * 10.0f) + 5.0f;
+    this->actor.speedXZ = 0.0f;
+    this->actor.posRot.rot.y = this->actor.shape.rot.y;
+    EnRd_SetupAction(this, func_80AE2744);
+}
+
+void func_80AE2744(EnRd* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    Math_SmoothScaleMaxMinS(&this->unk_30E, 0, 1, 0x64, 0);
+    Math_SmoothScaleMaxMinS(&this->unk_310, 0, 1, 0x64, 0);
+
+    if ((this->actor.params == 2) && (0.0f == this->skelAnime.animCurrentFrame)) {
+        if (Math_Rand_ZeroOne() >= 0.5f) {
+            SkelAnime_ChangeAnimDefaultRepeat(&this->skelAnime, &D_06005D98);
+        } else {
+            SkelAnime_ChangeAnimDefaultRepeat(&this->skelAnime, &D_060057AC);
+        }
+    } else {
+        this->unk_30C--;
+        if (this->unk_30C == 0) {
+            this->unk_30C = (Math_Rand_ZeroOne() * 10.0f) + 10.0f;
+            this->skelAnime.animCurrentFrame = 0.0f;
+        }
+    }
+
+    if (this->actor.parent != NULL) {
+        if (this->unk_305 == 0) {
+            if (this->actor.params != 2) {
+                func_80AE31DC(this);
+            } else {
+                func_80AE392C(this);
+            }
+        }
+    } else {
+        if (this->unk_305 != 0) {
+            if (this->actor.params != 2) {
+                func_80AE37BC(this);
+            } else {
+                func_80AE392C(this);
+            }
+        }
+
+        this->unk_305 = 0;
+        if (this->actor.xzDistFromLink <= 150.0f) {
+            if (func_8002DDE4(globalCtx) != 0) {
+                if ((this->actor.params != 2) && (this->unk_305 == 0)) {
+                    func_80AE37BC(this);
+                } else {
+                    func_80AE392C(this);
+                }
+            }
+        }
+    }
+
+    if ((globalCtx->gameplayFrames & 0x5F) == 0) {
```

Comment:
```suggestion
    if ((globalCtx->gameplayFrames % 96) == 0) {
```
I like `%` better but it seems to match inconsistently, I don't know if that would be preferred either?

Reply:
It doesn't work here, unfortunately. I changed the numbers to decimal though